overlord/ifacestate/handlers.go

```diff
@@ -458,13 +444,11 @@ func (m *InterfaceManager) doDisconnect(task *state.Task, _ *tomb.Tomb) error {
 		return err
 	}
 
-	affectedConns, err := m.repo.ResolveDisconnect(plugRef.Snap, plugRef.Name, slotRef.Snap, slotRef.Name)
+	err = m.repo.Disconnect(plugRef.Snap, plugRef.Name, slotRef.Snap, slotRef.Name)
 	if err != nil {
-		return err
+		return fmt.Errorf("snapd changed, please retry the operation: %v", err)
```

Comment:
I fear this is a bit more complicated, given that we don't use conflicts, we can get this here when plugRef.Snap or slotRef.Snap don't exist anymore at all, in which case the old code would just skip, see below, I think we need to change order of things around do the snapstate.Gets first, and then this, otherwise we would produce this error misleadingly in some corner cases